mat64/vector.go

```diff
@@ -357,6 +360,61 @@ func (v *Vector) MulVec(a Matrix, b *Vector) {
 	}
 }
 
+// MarshalBinary encodes the receiver into a binary form and returns the result.
+//
+// Vector is little-endian encoded as follows:
+//   0 -  8  number of elements     (int64)
+//   8 - ..  vector's data elements (float64)
+func (v Vector) MarshalBinary() ([]byte, error) {
+	buf := bytes.NewBuffer(make([]byte, 0, sizeInt64+v.n*sizeFloat64))
+	err := binary.Write(buf, defaultEndian, int64(v.n))
+	if err != nil {
+		return nil, err
+	}
+
+	for _, val := range v.mat.Data {
```

Comment:
Dense has the same problem. It doesn't marshal all of the data, just what is returned by At. Dense.MarshalBinary can be considered as doing 
````
r,c := d.Dims()
for i := 0; i < r; i++ {
   for j := 0; j < c; j++ {
        marshal(d.At(i,j))
   }
}
````
and similarly for a Vector
````
for i := 0; i < v.Len(); i++ {
    marshal(v.At(i,0))
}
````

I agree there are good uses for marshaling code that is hysteresis free. Gob would be a good candidate for using a marshaling that is specific for the internal representation. MarshalBinary is the best candidate for marshaling in a way that is clear and easy to use cross-language. The best way to keep it easily cross-platform is to only marshal the viewable data in the type.

I'm not exactly sure what your example illustrates. Why would the user want to individually track the inc of the vector? Non-unity incs only arise in mat64 when there is a ColView of a Dense. Knowing the number of columns in the Dense is not sufficient to know the Inc, because the user also has to know the Dense has never been viewed.